src/cleanup/security-group-cleaner.ts

```diff
@@ -79,3 +87,52 @@ function arrayContains(arr: string[], set: Set<string>): boolean {
 function filterOut(arr: string[], set: Set<string>): string[] {
   return arr.filter(el => !set.has(el));
 }
+export async function CleanupElasticacheSecurityGroups(
+  groupIds: Set<string>
+): Promise<void> {
+  const CacheClusters = await getRawCacheClusters();
+  const ReplicationGroups = await getRawReplicationGroups();
+
+  for (const ReplicationGroup of ReplicationGroups) {
+    await cleanReplicationGroup(ReplicationGroup, CacheClusters, groupIds);
+  }
+}
+
+async function cleanReplicationGroup(
+  replicationGroup: ReplicationGroup,
+  CacheClusters: CacheCluster[],
+  groupIds: Set<string>
+): Promise<void> {
+  if (
+    replicationGroup.NodeGroups === undefined ||
+    replicationGroup.NodeGroups[0] === undefined ||
+    replicationGroup.NodeGroups[0].NodeGroupMembers === undefined ||
+    replicationGroup.NodeGroups[0].NodeGroupMembers[0] === undefined
+  )
+    return;
+  const exampleCacheCluster =
+    replicationGroup.NodeGroups[0].NodeGroupMembers[0].CacheClusterId;
+  const cacheSecurityGroups = CacheClusters.find(
+    cache => cache.CacheClusterId === exampleCacheCluster
+  );
+  if (
+    cacheSecurityGroups === undefined ||
+    cacheSecurityGroups.SecurityGroups === undefined
+  )
+    return;
+  const cacheSecurityGroupsIds: string[] = [
+    ...cacheSecurityGroups.SecurityGroups.flatMap(
+      group => group.SecurityGroupId ?? ' '
+    ).filter(id => id !== ' ' && groupIds.has(id)),
```

Comment:
It looks like this expression filters security groups down to `groupIds`.

Please, confirm that it removes `groupIds` from the list of security groups instead of removing all the security groups except `groupIds`.